### ISSUE TITLE
Missing Layers after App is Backgrounded

### ISSUE THREAD

**Issue by riastrad** [mapbox/mapbox-gl-native#13899]
In an app with a significant amount data applied at runtime, when the app is backgrounded for a while there will occasionally be a background launch that triggers a memory warning. When the app is brought back to the foreground after this behavior several map layers are no longer being rendered. 

One potentially important detail here is that the app is using the [Significant Location Changes](https://developer.apple.com/documentation/corelocation/cllocationmanager/1423531-startmonitoringsignificantlocati) API, which is triggering an app launch and subsequent memory warnings if a user moves more than 500 meters while the app is backgrounded. 

### Steps to reproduce

_Note that this behavior is tricky to consistently replicate and may take several tries_

 1. Open app and observe that map loads correctly with all layers intact
 2. Background the app and move around in order to trigger the app relaunch
 3. Return to the app and notice that the layers are no longer showing

### Expected behavior

Layers should load correctly, like they do when app is initially launched.

<img width="189" alt="simulator screen shot - iphone xr - 2019-02-08 at 11 25 09" src="https://user-images.githubusercontent.com/16272584/52491469-2f962000-2b95-11e9-84b6-79aa73ecd793.png" />

### Actual behavior

Most (but not all) map layers disappear from the app.

<img width="189" alt="screen shot 2019-02-08 at 11 32 39 am" src="https://user-images.githubusercontent.com/16272584/52491574-6cfaad80-2b95-11e9-8728-7b9964d6bc75.png">

### Current hypothesis

When the app is in the background [`MGLRenderFrontend::render()`](https://github.com/mapbox/mapbox-gl-native/blob/f3341dd589c6330c0cfd6e8e381398d08493b48a/platform/darwin/src/MGLRendererFrontend.h#L47-L53) isn't running (and [`TilePyramid.update()`](https://github.com/mapbox/mapbox-gl-native/blob/ac3d59862a6632de0cd5076ba2a65f4e4a4577f4/src/mbgl/renderer/tile_pyramid.hpp#L36-L44) isn't called). When a memory warning is triggered `TilePyramid`'s cache is cleared. It's possible that nothing is holding on to a Tile, so when the app is launched, that data is missing.

### Configuration

**Mapbox SDK versions:** 
**iOS/macOS versions:** observed on an iPhone 7 & XS (latest iOS)
**Device/simulator models:**
**Xcode version:** 10.1

cc: @mapbox/maps-ios

**Comment by jjhampton** [comment]
Hi, I'm also seeing this on an application using MapBox for iOS. Is there any update on this?  cc @fabian-guerra

**Comment by julianrex** [comment]
@jjhampton did you see memory warnings before this happened to you? Can you reproduce it?

**Comment by jjhampton** [comment]
@julianrex No, I don't see any warnings, looking through the XCode debugger logs. However it occurs roughly 1 out of every 4 times that I background the app, and then return it to the foreground.

**Comment by fvonk** [comment]
We have the same issue in mapbox ios sdk 5.5.0 version. Is there any solution for this ? 
Thanks in advance.
